ryu/d2fixed.c

```diff
@@ -118,6 +118,10 @@ static inline uint32_t mulShift(const uint64_t m, const uint64_t* const mul, con
   const uint64_t s1low = high1 + low2 + c1; // 128
   const uint64_t c2 = s1low < high1;
   const uint64_t s1high = high2 + c2;       // 192
+  // If this assertion fails, c2 has been calculated incorrectly.
+  // This is possible for general multiplications,
```

Comment:
This is unclear to me. In what case can this happen?

Reply:
What if the first addition carries, so c1 is 1, and then low2 is all F hexits? Then low2 + c1 is 0 (emitting a carry). So s1low will be equal to high1, and c2 will be 0, even though it should be 1 to propagate the carry.

Does something about multiplication prevent this scenario from happening? I didn’t backtrack the math to construct an explicit counterexample.

Reply:
Good catch. I don't immediately see any reason why it couldn't happen. It may be rare enough that we don't see it with the existing tests.